File: leetcode/arrays/(easy) Good Pairs/solves.py
# My solution in python:
# Memory usage:   bad (above 27% of py submissions)
# Runtime:        good (above 72% of py submissions)
# Time complex.:  O(n²) - Traverses the array twice
# Space complex.: O(1+m) - Stores a dict with m keys where m is the number of distinct elements

class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        c=0
        for i in range(0,len(nums)-1):
            for j in range((i+1),len(nums)):
                if nums[i]==nums[j]:
                    c+=1
        return c

#PS (post submission):
# 1. Python for loops are so sad :(

File: leetcode/arrays/(easy) Good Pairs/solves.js
/*
My solution:
Memory usage:   ok (above 65% of js submissions)
Runtime:        ok (above 63% of js submissions)
Time complex.:  O(n²) - Traverses the array twice
Space complex.: O(1) - Doesn't store anything new
*/

/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function(nums) {
    let n_pairs = 0;
    let n = nums.length;

    for (let i = 0; i<n; i++){
        for (let j = i+1; j<n; j++){
            if (nums[i] === nums[j]) {
                n_pairs++
            }
        }
    }
    return n_pairs;
};
